skada/transformers.py

```diff
@@ -0,0 +1,66 @@
+# Author: Remi Flamary <remi.flamary@polytechnique.edu>
+#
+# License: BSD 3-Clause
+
+from sklearn.utils import check_random_state
+
+from .base import BaseAdapter
+from .utils import check_X_y_domain
+
+
+class SubsampleTransformer(BaseAdapter):
+    """Transformer that subsamples the data.
+
+    This transformer is useful to speed up computations when the data is too
+    large. It randomly selects a subset of the data to work with during training
+    but does not change the data during testing.
+
+    .. note::
+        This transformer should not be used as the last step of a pipeline
+        because it returns non standard output.
+
+    Parameters
+    ----------
+    n_subsample : int
+        Number of samples to keep (keep all if data smaller).
+    random_state : int, RandomState instance or None, default=None
+        Controls the random resampling of the data.
+    """
+
+    def __init__(self, n_subsample, random_state=None):
+        self.n_subsample = n_subsample
+        self.random_state = random_state
+
+    def _pack_params(self, idx, **params):
+        return {
+            k: (v[idx] if idx is not None else v)
+            for k, v in params.items()
+            if v is not None
+        }
+
+    def fit_transform(self, X, y=None, *, sample_domain=None, sample_weight=None):
```

Comment:
I guess it would be nice to have test case to check if `sample_weight` is processed properly, as right now we only have checks in place for `sample_domain`